File: web/src/features/platform/api/platformApi.js
import api from "../../../core/api/axios";

export const platformApi = {
  list: () => api.get("/platforms"),
  create: (name) => api.post("/platforms", { name }),
  update: (id, name) => api.put(`/platforms/${id}`, { name }),
  remove: (id) => api.delete(`/platforms/${id}`),
};

File: mobile/app/src/main/java/edu/cit/capendit/unisell/platform/api/PlatformApi.kt
package edu.cit.capendit.unisell.platform.api

import edu.cit.capendit.unisell.platform.model.PlatformRequest
import edu.cit.capendit.unisell.platform.model.PlatformResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.PUT
import retrofit2.http.POST
import retrofit2.http.Path

interface PlatformApi {

    @GET("platforms")
    suspend fun getPlatforms(): Response<List<PlatformResponse>>

    @POST("platforms")
    suspend fun createPlatform(@Body request: PlatformRequest): Response<PlatformResponse>

    @PUT("platforms/{id}")
    suspend fun updatePlatform(
        @Path("id") id: Long,
        @Body request: PlatformRequest
    ): Response<PlatformResponse>

    @DELETE("platforms/{id}")
    suspend fun deletePlatform(@Path("id") id: Long): Response<Unit>
}